geonode/layers/views.py

```diff
@@ -653,6 +654,58 @@ def layer_thumbnail(request, layername):
             )
 
 
+def layer_list(request):
+    """Get Layer list as JSON
+    """
+    if request.method == 'GET':
+        if 'q' in request.GET:
+            query = request.GET.get('q')
+
+            layers = Layer.objects.filter(
+                name__istartswith=query)
+            result = []
+            for layer in layers:
+                result.append(layer.name)
+        else:
+            layers = Layer.objects.all()
+            result = []
+            for layer in layers:
+                result.append(layer.name)
+        result = json.dumps(result)
+        return HttpResponse(result, content_type='application/json')
+
+
+def get_layer(request, layername):
```

Comment:
Shouldnt we just use the existing API for this? http://demo.geonode.org/api/layers/1/